### ISSUE TITLE
New Rule: Prefer mock resolved/rejected shorthands

### ISSUE THREAD

**Issue by alexilyaev** [jest-community/eslint-plugin-jest#104]
> Simple sugar function for:
> 
> ```js
> jest.fn().mockReturnValue(Promise.resolve(value));
> ```
>
> Useful to mock async functions in async tests:
> 
> ```js
> test('async test', async () => {
>   const asyncMock = jest.fn().mockResolvedValue(43);
> 
>   await asyncMock(); // 43
> });
> ```

Refs:

- https://facebook.github.io/jest/docs/en/mock-function-api.html#mockfnmockresolvedvaluevalue
- https://facebook.github.io/jest/docs/en/mock-function-api.html#mockfnmockresolvedvalueoncevalue
- https://facebook.github.io/jest/docs/en/mock-function-api.html#mockfnmockrejectedvaluevalue
- https://facebook.github.io/jest/docs/en/mock-function-api.html#mockfnmockrejectedvalueoncevalue

**Comment by SimenB** [comment]
Yeah, sounds good!

**Comment by G-Rath** [comment]
fwiw this is actually technically not the same due to https://github.com/facebook/jest/issues/6645 but I still think it's worth a rule with a fixer to start with, and turn it into a suggestion if we get complaints.

(I've so far only hit this once (https://github.com/slorber/react-async-hook/issues/24) so am hoping it's very rare).
